### PR TITLE
refactor(k8s, client): update kubernetes client as per idiomatic maya

### PR DESCRIPTION
**Pull request by AmitKumarDas**
This commit handles following idiomatic practices:
- reduce lines of code within a function
- code and code comments are well within 80 chars
- rename functions based on their behaviour

Signed-off-by: AmitKumarDas <amit.das@mayadata.io>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: pkg/kubernetes/client/v1alpha1/client.go [hunk 1]
```diff
@@ -28,199 +28,252 @@ import (
...
-// buildConfigFromFlagsFunc provides the abstraction to get
-// kubernetes config from provided flags
+// buildConfigFromFlagsFn is a typed function
+// to abstract getting a kubernetes config from
+// provided flags
 //
 // NOTE:
 //  typed function makes it simple to mock
-type buildConfigFromFlagsFunc func(string, string) (*rest.Config, error)
+type buildConfigFromFlagsFn func(string, string) (*rest.Config, error)
```
Comment (sonasingh46) [inline pkg/kubernetes/client/v1alpha1/client.go]: What is the idea behind this change ?  `Func` ----? `Fn`

File: pkg/kubernetes/client/v1alpha1/client.go [hunk 2]
```diff
@@ -28,199 +28,252 @@ import (
...
-	// K8sMasterIPEnvironmentKey is the environment variable key used to
-	// determine the kubernetes master IP address
+	// K8sMasterIPEnvironmentKey is the environment variable
+	// key used to provide kubernetes master IP address
 	K8sMasterIPEnvironmentKey env.ENVKey = "OPENEBS_IO_K8S_MASTER"
 
-	// KubeConfigEnvironmentKey is the environment variable key used to
-	// determine the kubernetes config
+	// KubeConfigEnvironmentKey is the environment variable
+	// key used to provide kubernetes config
```
Comment (sonasingh46) [inline pkg/kubernetes/client/v1alpha1/client.go]: is this the path for `KUBEONFIG` ? What is the value of the key here?

Comment (sonasingh46) [reply pkg/kubernetes/client/v1alpha1/client.go]: I guess it is kubeconfig path ... Can we specify that ?

File: pkg/kubernetes/client/v1alpha1/client.go [hunk 3]
```diff
@@ -28,199 +28,252 @@ import (
...
+}
+
+var defaultFns = []OptionFn{
+	withDefaultGetInClusterConfigFn(),
+	withDefaultBuildConfigFromFlagsFn(),
+	withDefaultGetKubeClientsetFn(),
+	withDefaultGetKubeDynamicClientFn(),
+	withDefaultGetKubeMasterIPFromENVFn(),
+	withDefaultGetKubeConfigPathFromENVFn(),
+}
```
Comment (sonasingh46) [inline pkg/kubernetes/client/v1alpha1/client.go]: If we get in cluster config from `withDefaultGetInClusterConfigFn`  -- Do we still require to execute other `withs...`?

File: pkg/kubernetes/client/v1alpha1/client.go (hunk 1 above)
Comment (AmitKumarDas) [reply pkg/kubernetes/client/v1alpha1/client.go]: Func vs. Fn both seem appropriate to convey their purpose.
I chose `Fn` as it is shorter.
When the actual function names are large `Fn` when used as a suffix helps.

File: pkg/kubernetes/client/v1alpha1/client.go (hunk 2 above)
Comment (AmitKumarDas) [reply pkg/kubernetes/client/v1alpha1/client.go]: I will do something to reduce this confusion. Thanks.

File: pkg/kubernetes/client/v1alpha1/client.go (hunk 3 above)
Comment (AmitKumarDas) [reply pkg/kubernetes/client/v1alpha1/client.go]: No required.

However, this is a dumb list of settings that sets the default options.
Caller code need not bother about it at all.
There is no change in functionality when compared with older version with this version.
This version make the logic modular & also makes LOC in withDefaults less.

File: pkg/kubernetes/client/v1alpha1/client.go (hunk 2 above)
Comment (AmitKumarDas) [reply pkg/kubernetes/client/v1alpha1/client.go]: done

**Pull request merged**